lib/doorkeeper/models/concerns/hashable.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module Models
+    ##
+    # Hashable finder to provide lookups for input plaintext values which are
+    # mapped to their hashes before lookup.
+    module Hashable
+      extend ActiveSupport::Concern
+
+      module ClassMethods
+        private
+
+        # Returns an instance of the Doorkeeper::AccessToken with
+        # specific token value.
+        #
+        # @param attr [Symbol]
+        #   The token attribute we're looking with.
+        #
+        # @param token [#to_s]
+        #   token value (any object that responds to `#to_s`)
+        #
+        # @return [Doorkeeper::AccessToken, nil] AccessToken object or nil
+        #   if there is no record with such token
+        #
+        def find_by_hashed_or_plain_token(attr, token)
```

Comment:
That is absolutely fine with me. 👍

Reply:
Looking at the code again, your helpers already exist at the model level. I believe the `Hashable` concern is correct and should not know details about what attributes exist (Otherwise you'll have `find_by_refresh_token` in the concern resulting in it existing in the access grant model which doesn't know refresh tokens.

The places you're looking for are those two, they should be extended exactly the same like in your current setup.
https://github.com/doorkeeper-gem/doorkeeper/pull/1168/files#diff-dc86e93a23661f9a5d87d2b031af8ba5R34
https://github.com/doorkeeper-gem/doorkeeper/pull/1168/files#diff-c42935829979fd1bcb5ed418c92f6386R26